client/src/components/ProfileCard/ProfileCard.tsx

```diff
@@ -38,53 +40,58 @@ const Status = styled(Typography)(({ theme }) => ({
   color: theme.palette.grey[500],
 }));
 
-const Description = styled(Typography)(({ theme }) => ({
+const Description = styled(LinesEllipsis)(({ theme }) => ({
   ...theme.typography.body1,
   textAlign: 'center',
   fontSize: 16,
   fontWeight: 400,
+  height: 50,
+  maxHeight: 50,
 }));
 
 const Location = styled(Typography)(({ theme }) => ({
   ...theme.typography.body1,
   color: theme.palette.grey[500],
 }));
 
-interface Props {
-  id: string;
+interface Props extends Omit<Profile, 'userId' | 'isSitter' | 'gender' | 'telephone' | 'birthday' | 'uploadedImages'> {
+  hourlyRate: number;
 }
 
-const ProfileCard: FunctionComponent<Props> = ({ id }) => {
+const ProfileCard: FunctionComponent<Props> = (profile) => {
   return (
     <StyledCard variant="elevation">
-      <CardActionArea component={Link} to={`profile-details/${id}`} disableRipple>
+      <CardActionArea component={Link} to={`profile-details/${profile.id}`} disableRipple>
         <CardContent sx={{ p: 0 }}>
           <Stack alignItems="center" gap={1} p={2} pt={4}>
-            <ProfileAvatar src="https://robohash.org/hello-world" />
+            <ProfileAvatar src={profile.photo || `https://robohash.org/${profile.id}`} />
             <Stack alignItems="center">
-              <Title>Norma Byers</Title>
-              <Status>Loving pet sitter</Status>
+              <Title>{profile?.name || 'No Name'}</Title>
+              <Status>{profile?.jobTitle || 'Unknown'}</Status>
               <Rating
                 name="rating"
                 readOnly
-                value={3}
+                value={profile?.averageRating || 0}
                 size="small"
                 emptyIcon={<Star style={{ opacity: 0.55 }} fontSize="inherit" />}
               />
             </Stack>
-            <Description variant="body1" fontSize={14}>
-              Dog sitting, cat sitting, pocket pet and bird care
-            </Description>
+            <Description
+              text={profile?.description || 'No description provided'}
+              maxLine="2"
+              basedOn="letters"
+              trimRight
+            />
           </Stack>
           <Box pt={3}>
             <Divider />
             <Stack direction="row" justifyContent="space-between" alignItems="center" p={3}>
               <Stack direction="row" spacing={0.5}>
                 <LocationOn color="primary" />
-                <Location>Toronto, Ontario</Location>
+                <Location>{profile?.address || 'No location'}</Location>
```

Comment:
Since we search for sitter profiles based on availability and location, let's avoid displaying any profiles which don't have a location

Reply:
I definitely can, it's just that this profile listings page, although receives the "location" , "start" and "end" date. Because the availability was not complete at the time of integration, It does not actually filter user's based off of the provided query information.